libcxx/include/vector

```diff
@@ -725,8 +726,7 @@ public:
 private:
   pointer __begin_ = nullptr;
   pointer __end_   = nullptr;
-  __compressed_pair<pointer, allocator_type> __end_cap_ =
-      __compressed_pair<pointer, allocator_type>(nullptr, __default_init_tag());
+  _LIBCPP_COMPRESSED_PAIR(pointer, __cap_, allocator_type, __alloc_);
```

Comment:
You should be able to restore the `= nullptr` here and re-simplify the ctors after my suggestion to the macro.